### ISSUE TITLE
Spyder crashes showing tooltip while typing

### ISSUE THREAD

**Issue by penguinpee** [spyder-ide/spyder#18011]
### What steps will reproduce the problem?

Just typing in the editor. Tooltip briefly shows and then issue reporter pops up.

I'm using a local module installed with `pip`. Maybe there's an issue with the module.

### Traceback
```python-traceback
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/spyder/plugins/editor/widgets/codeeditor.py", line 3011, in show_hint_for_completion
    self.show_hint(
  File "/usr/lib/python3.10/site-packages/spyder/widgets/mixins.py", line 641, in show_hint
    self.show_tooltip(signature=html_signature, text=extra_text,
  File "/usr/lib/python3.10/site-packages/spyder/widgets/mixins.py", line 620, in show_tooltip
    self.tooltip_widget.show_tip(point, tiptext, cursor=cursor,
  File "/usr/lib/python3.10/site-packages/spyder/widgets/calltip.py", line 192, in show_tip
    point.setY(adjusted_point.y() - tip_height - padding)
TypeError: setY(self, int): argument 1 has unexpected type 'float'
```

## Versions

* Spyder version: 5.1.5 
* Python version: 3.10.4
* Qt version: 5.15.2
* PyQt5 version: 5.15.6
* Operating System: Linux 5.17.8-200.fc35.x86_64

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  1.6.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  7.26.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.0 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  23.5.0 (OK)
nbconvert >=4.0               :  6.1.0 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
parso >=0.7.0;<0.9.0          :  0.8.2 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.9.0 (OK)
pylint >=2.5.0                :  2.9.6 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.2.2;<1.3.0          :  1.2.2 (OK)
pylsp_black >=1.0.0           :  None (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.2.0 (OK)
qtawesome >=1.0.2             :  1.0.3 (OK)
qtconsole >=5.1.0             :  5.1.1 (OK)
qtpy >=1.5.0                  :  1.11.2 (OK)
rtree >=0.9.4                 :  0.9.4 (OK)
setuptools >=49.6.0           :  57.4.0 (OK)
sphinx >=0.6.6                :  4.1.2 (OK)
spyder_kernels >=2.1.1;<2.2.0 :  2.1.1 (OK)
textdistance >=4.2.0          :  4.2.0 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.6 (OK)
xdg >=0.26                    :  0.27 (OK)
zmq >=17                      :  22.2.1 (OK)

# Optional:
cython >=0.21                 :  None (OK)
matplotlib >=2.0.0            :  3.5.2 (OK)
numpy >=1.7                   :  1.21.5 (OK)
pandas >=1.1.1                :  1.3.3 (OK)
scipy >=0.17.0                :  1.8.0 (OK)
sympy >=0.7.3                 :  None (OK)
```

**Comment by ccordoba12** [comment]
Hey @penguinpee, thanks for reporting. This is fixed in our latest version (5.3.1), so please update to that version.

We don't recommend you to use `sudo pip install` for that because it could break your system Python. Instead, please use a virtualenv.

**Comment by penguinpee** [comment]
> Hey @penguinpee, thanks for reporting. This is fixed in our latest version (5.3.1), so please update to that version.

Thanks. I'll update.
 
> We don't recommend you to use `sudo pip install` for that because it could break your system Python. Instead, please use a virtualenv.

To clarify, Spyder has been installed from Fedora 35 repository. I was working with a local module installed with `pip` when the error occurred. But apparently that's not the issue.

**Comment by ccordoba12** [comment]
> To clarify, Spyder has been installed from Fedora 35 repository

Yep, I saw that.

> I was working with a local module installed with pip when the error occurred. But apparently that's not the issue.

No, the problem is that the Spyder version distributed by Fedora is not totally compatible with Python 3.10. And I just mentioned you the warning about `sudo pip install` because many people new to Linux just blindly use it to update to a Spyder version that doesn't come with their distros.
